### ISSUE TITLE
Revisit total balance to include more tokens and token order

### ISSUE THREAD

**Issue by fbwoolf** [blockstack/blockstack-app#5159]
As we start to add more fiat values to our assets list, we need to revisit how our code is setup to add new token values to our total fiat balance shown in the account header.

As part of this work, we should look at refactoring how we handle metadata for Stacks FTs. It was requested that the balances be in descending fiat balance order, but with how we handle balances and fiat prices (market data), it is difficult to do this with a larger refactor.

See comments on this issue for ref: https://github.com/leather-wallet/extension/pull/5151#issuecomment-2034182522

**Comment by fbwoolf** [comment]
@markmhendrickson it might be that we want to prioritize this refactoring now that we have move fiat values getting pushed?

**Comment by kyranjamie** [comment]
+1. 

How we read balance data in general needs a refactor. Suggest the solution to this considers how we can first collect all relevant data, then render the UI, rather than individual components fetching their own data.

We also need to be very alert caching/request efficiency in solving this.

**Comment by markmhendrickson** [comment]
Yep agreed on making this a priority. @fbwoolf thanks for taking it
